src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1508,7 +1508,7 @@ HashSet<string> GetPackageDependencies()
                     return results;
                 }
 
-                static string GetPackageId(LockFileTargetLibrary package) => $"{package.Name}/{package.Version.ToNormalizedString()}";
+                static string GetPackageId(LockFileTargetLibrary package) => $"{package.Name}/{package.Version.ToNormalizedString().ToLowerInvariant()}";
```

Comment:
Same thing probably applies here. 
The checks using this string are ordinal ignore case.